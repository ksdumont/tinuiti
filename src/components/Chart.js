import React from 'react'
import {Line} from 'react-chartjs-2';

function Chart(props) {
const state = {
  labels: ['January', 'February', 'March',
           'April', 'May'],
  datasets: [
    {
      label: 'Rainfall',
      fill: false,
      lineTension: 0.5,
      backgroundColor: '#6600cc',
      borderColor: '#4178cf',
      borderWidth: 2,
      data: [65, 59, 80, 81, 56]
    }
  ]
}

  return (
    <div className="chart-container">
        <Line
          data={state}
          options={{
            title:{
              display:true,
              text:'Average Rainfall per month',
              fontSize:20
            },
            legend:{
              display:true,
              position:'right'
            }
          }}
        />
      </div>
  )
}

export default Chart
