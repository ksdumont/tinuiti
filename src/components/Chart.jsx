import React from 'react';
import { Line } from 'react-chartjs-2';
import PropTypes from 'prop-types';

function Chart({ keyword, dailyRankings }) {
  const chartData = {
    labels: ['10/01/20', '10/02/20', '10/03/20', '10/04/20', '10/05/20'],
    datasets: [
      {
        label: 'SEO Rankings',
        fill: true,
        lineTension: 0.5,
        backgroundColor: 'rgba(102,0,204, 0.5)',
        borderColor: '#4178cf',
        borderWidth: 5,
        data: dailyRankings === undefined ? [] : dailyRankings.map((r) => r.rank),
      },
    ],
  };

  return (
    <div className="chart-container">
      <Line
        data={chartData}
        options={{
          title: {
            display: true,
            text: `Keywords: ${keyword}`,
            fontSize: 20,
          },
          legend: {
            display: true,
            position: 'right',
          },
        }}
      />
    </div>
  );
}

Chart.propTypes = {
  keyword: PropTypes.string,
  dailyRankings: PropTypes.instanceOf(Array),
};
Chart.defaultProps = {
  keyword: 'tempor aliquip enim',
  dailyRankings: [{
    rank: 40,
    date: '2020-10-01',
  },
  {
    rank: 48,
    date: '2020-10-02',
  },
  {
    rank: 32,
    date: '2020-10-03',
  },
  {
    rank: 21,
    date: '2020-10-04',
  },
  {
    rank: 17,
    date: '2020-10-05',
  }],
};

export default Chart;
