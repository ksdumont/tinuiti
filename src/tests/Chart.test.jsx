import React from 'react';
import ReactDOM from 'react-dom';
import Chart from '../components/Chart';

import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const wrapper = shallow(<Chart />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
