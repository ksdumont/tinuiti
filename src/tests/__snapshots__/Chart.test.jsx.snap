// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="chart-container"
>
  <Line
    data={
      Object {
        "datasets": Array [
          Object {
            "backgroundColor": "rgba(102,0,204, 0.5)",
            "borderColor": "#4178cf",
            "borderWidth": 5,
            "data": Array [
              40,
              48,
              32,
              21,
              17,
            ],
            "fill": true,
            "label": "SEO Rankings",
            "lineTension": 0.5,
          },
        ],
        "labels": Array [
          "10/01/20",
          "10/02/20",
          "10/03/20",
          "10/04/20",
          "10/05/20",
        ],
      }
    }
    options={
      Object {
        "legend": Object {
          "display": true,
          "position": "right",
        },
        "title": Object {
          "display": true,
          "fontSize": 20,
          "text": "Keywords: tempor aliquip enim",
        },
      }
    }
  />
</div>
`;
